src/dune_rules/inline_tests.ml

```diff
@@ -229,51 +312,31 @@ include Sub_system.Register_end_point (struct
         in
         Super_context.add_alias_action sctx ~dir ~loc:(Some info.loc)
           (Alias.make ~dir runtest_alias)
-          (let exe =
-             Path.build (Path.Build.relative inline_test_dir (name ^ ext))
-           in
-           let open Action_builder.O in
-           let deps, sandbox =
-             let sandbox =
-               let project = Scope.project scope in
-               if Dune_project.dune_version project < (3, 5) then
-                 Sandbox_config.no_special_requirements
-               else Sandbox_config.needs_sandboxing
-             in
-             Dep_conf_eval.unnamed ~sandbox info.deps ~expander
-           in
-           let+ () = deps
-           and+ () = Action_builder.paths source_files
-           and+ () = Action_builder.path exe
-           and+ action =
-             match custom_runner with
-             | None ->
-               let+ flags = flags in
-               Action.run (Ok exe) flags
-             | Some runner -> (
-               let* prog =
-                 Action_builder.of_memo
-                   (Super_context.resolve_program ~dir sctx ~loc:(Some loc)
-                      runner)
-               and* flags = flags in
-               let action =
-                 Action.run prog (Path.reach exe ~from:(Path.build dir) :: flags)
-               in
-               (* jeremiedimino: it feels like this pattern should be pushed
-                  into [resolve_program] directly *)
-               match prog with
-               | Error _ -> Action_builder.return action
-               | Ok p -> Action_builder.path p >>> Action_builder.return action)
-           in
-           let run_tests = Action.chdir (Path.build dir) action in
-           Action.Full.make ~sandbox
-           @@ Action.progn
-                (run_tests
-                :: List.map source_files ~f:(fun fn ->
-                       Action.diff ~optional:true fn
-                         (Path.Build.extend_basename
-                            (Path.as_in_build_dir_exn fn)
-                            ~suffix:".corrected")))))
+          (let open Action_builder.O in
+          let+ actions =
+            let* partitions_flags =
+              match partitions_flags with
+              | None -> Action_builder.return [ None ]
+              | Some _ ->
+                let+ partitions =
+                  Action_builder.lines_of (Path.build partition_file)
+                in
+                List.map ~f:(fun x -> Some x) partitions
+            in
+            List.map partitions_flags ~f:(fun p -> action mode (flags p))
```

Comment:
Is it intended to ask for all these deps to just compute the partitions file?

Also, should we just sandbox the computation of partitions by default?

Reply:
> Is it intended to ask for all these deps to just compute the partitions file?

What do you mean by `all these deps` ?

> Also, should we just sandbox the computation of partitions by default?

I don't know much about sandboxing in dune, How does one change the default ?

Reply:
> What do you mean by all these deps ?

I mean `info.deps` - the deps that are in `(inline_tests (deps ..))`

> I don't know much about sandboxing in dune, How does one change the default ?

I can do this.

Reply:
You could imagine that the content of some info.deps is used to compute the list of partitions.